src/module.c

```diff
@@ -3783,21 +3783,27 @@ static void moduleInitKeyTypeSpecific(RedisModuleKey *key) {
  * The return value is the handle representing the key, that must be
  * closed with RM_CloseKey().
  *
- * If the key does not exist and WRITE mode is requested, the handle
+ * If the key does not exist and REDISMODULE_WRITE mode is requested, the handle
  * is still returned, since it is possible to perform operations on
  * a yet not existing key (that will be created, for example, after
- * a list push operation). If the mode is just READ instead, and the
+ * a list push operation). If the mode is just REDISMODULE_READ instead, and the
  * key does not exist, NULL is returned. However it is still safe to
  * call RedisModule_CloseKey() and RedisModule_KeyType() on a NULL
- * value. */
+ * value.
+ *
+ * Extra flags that can be pass to the API under the mode argument:
+ * * REDISMODULE_OPEN_KEY_NOTOUCH - Avoid touching the LRU/LFU of the key when opened.
+ * * REDISMODULE_OPEN_KEY_NONOTIFY - Don't trigger keyspace event on key misses.
+ * * REDISMODULE_OPEN_KEY_NOSTATS - Don't update keyspace hits/misses counters.
+ * * REDISMODULE_OPEN_KEY_NOEXPIRE - Avoid deleting lazy expired keys.
+ * * REDISMODULE_OPEN_KEY_NOEFFECTS - Avoid any effects from fetching the key. */
```

Comment:
maybe we should mention that it includes all the above, and possibly some future ones as well?

Reply:
@oranagra if a module with older .h file is using it then it will not include future ones no?

Reply:
good catch, so it'll be a source code future compatibility, but not binary future compatibility.
we can choose to document that, or maybe change that that define to be just another enum and map it to the other flags at runtime in module.c..

i think the second option is better.

Reply:
OK, so `NOEFFECTS` will be another value that will be checked on runtime and if in the future we will add more effect that happened as a result of RM_OpenKey we will know to avoid them :+1:

Reply:
Done